tensorboard/components/tf_data_selector/tf-data-select-row.ts

```diff
@@ -14,64 +14,189 @@ limitations under the License.
+    _tagSelectionStateString: {type: String, value: ''},
+
+    _selectedTags: {
+      type: Array,
+      value: () => [],
     },
   },
 
-  get _runsStore() {
-    return tf_backend.runsStore;
+  observers: [
+    '_persistSelectedRuns(_selectedRuns)',
+    '_persistSelectedTags(_selectedTags)',
+  ],
+
+  _getPersistenceKey(type: Type): string {
+    const number = this.persistenceNumber || 0;
+    switch (type) {
+      case Type.RUN:
+        return `g${number}r`;
+      case Type.TAG:
+        return `g${number}t`;
+    }
+  },
+
+  ready() {
+    if (this.persistenceNumber == null) return;
+
+    const runInitializer = tf_storage.getStringInitializer(
+        this._getPersistenceKey(Type.RUN),
+        {defaultValue: '', polymerProperty: '_runSelectionStateString'});
+    runInitializer.call(this);
+
+    const tagInitializer = tf_storage.getStringInitializer(
+        this._getPersistenceKey(Type.TAG),
+        {defaultValue: '', polymerProperty: '_tagSelectionStateString'});
+    tagInitializer.call(this);
   },
 
   attached() {
-    this._updateRunKey = this._runsStore.addListener(() => this._updateRuns());
-    this._updateRuns();
+    this._fetchRunsAndTags().then(() => this._isDataReady = true);
   },
 
   detached() {
-    this._runsStore.removeListenerByKey(this._updateRunKey);
+    this._isDataReady = false;
   },
 
-  _updateRuns() {
-    this.set('_runs', this._runsStore.getRuns());
+  _fetchRunsAndTags() {
+    const requestManager = new tf_backend.RequestManager();
+    if (this.noExperiment) {
+      const fetchRuns = requestManager.request(tf_backend.getRouter().runs());
+      return Promise.all([fetchRuns]).then(([runs]) => {
+        this.set('_runs', Array.from(new Set(runs)).map(runName => ({
+          id: runName,
+          name: runName,
+          startedTime: null,
+        })));
+      });
+    } else if (this.experiment.id) {
+      const url = tf_backend.getRouter().runsForExperiment(this.experiment.id);
+      return requestManager.request(url).then(runs => {
+        this.set('_runs', runs);
+        // Flatten the tags.
+        const tagSet = new Map();
+        runs.forEach(({tags}) => {
+          tags.forEach(tag => tagSet.set(tag.id, tag));
+        })
+        this.set('_tags', Array.from(tagSet.values()));
+      });
+    }
   },
 
-  _storeRunSelectionState:
-      tf_storage.getObjectObserver('runSelectionState', {
-        defaultValue: {},
-        polymerProperty: '_runSelectionState',
-      }),
+  _getRunOptions(_) {
+    return this._runs.map(run => ({
+      id: run.id,
+      title: run.name,
+    }));
+  },
 
-  _storeRunRegexInput:
-      tf_storage.getStringObserver('regexInput', {
-        defaultValue: '',
-        polymerProperty: '_runRegexInput',
-      }),
+  _getTagOptions(_) {
+    return this._tags.map(tag => ({
+      id: tag.id,
+      title: tag.name,
+    }));
+  },
+
+  _getIsRunCheckboxesColored(_) {
+    return this.noExperiment;
+  },
+
+  _computeSelection(_, __) {
```

Comment:
Is this just not wired up yet?  Leaving a TODO comment or something helps make it clear that it's intentional.